plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -139,19 +164,24 @@ class vtkF3DOCCTReader::vtkInternals
         linesCells->InsertNextCell(polyline.size(), polyline.data());
 
 #if F3D_PLUGIN_OCCT_XCAF
-        if (this->ColorTool)
+        std::array<unsigned char, 3> rgb = { 0, 0, 0 };
+        try
         {
-          std::array<unsigned char, 3> rgb = { 0, 0, 0 };
-          Quantity_Color aColor;
-          if (this->ColorTool->GetColor(edge, XCAFDoc_ColorCurv, aColor) ||
-            this->ColorTool->GetColor(shape, XCAFDoc_ColorCurv, aColor))
+          const auto& style = inheritedStyles.FindFromKey(edge);
+          if (style.IsSetColorCurv())
           {
-            rgb[0] = static_cast<unsigned char>(255.0 * aColor.Red());
-            rgb[1] = static_cast<unsigned char>(255.0 * aColor.Green());
-            rgb[2] = static_cast<unsigned char>(255.0 * aColor.Blue());
+            Quantity_Color color = style.GetColorCurv();
+            rgb[0] = static_cast<unsigned char>(255.0 * color.Red());
+            rgb[1] = static_cast<unsigned char>(255.0 * color.Green());
+            rgb[2] = static_cast<unsigned char>(255.0 * color.Blue());
           }
-          colors->InsertNextTypedTuple(rgb.data());
         }
+        catch (Standard_NoSuchObject&)
+        {
+          /* ignore */
```

Comment:
no need to warn or log ?

Reply:
no, just means the part has no style set, i'll update the comment